coderd/insights_test.go

```diff
@@ -2351,3 +2351,101 @@ func TestGenericInsights_RBAC(t *testing.T) {
 		})
 	}
 }
+
+func TestTotalUsersInsight(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Success", func(t *testing.T) {
+		t.Parallel()
+
+		client, db := coderdtest.NewWithDatabase(t, nil)
+		coderdtest.CreateFirstUser(t, client)
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
+		t.Cleanup(cancel)
+
+		// Given: a deployment with many users
+
+		dbgen.User(t, db, database.User{
+			Email:     "user1@coder.com",
+			Username:  "user1",
+			CreatedAt: daysAgo(2),
+		})
+		dbgen.User(t, db, database.User{
+			Email:     "user2@coder.com",
+			Username:  "user2",
+			CreatedAt: daysAgo(2),
+		})
+		dbgen.User(t, db, database.User{
+			Email:     "user3@coder.com",
+			Username:  "user3",
+			CreatedAt: daysAgo(1),
+		})
+
+		// When: requesting the accumulated total of users by day
+		res, err := client.TotalUsersInsight(ctx, codersdk.TotalUsersInsightRequest{
+			StartTime: daysAgo(2),
+			EndTime:   today(),
```

Comment:
It would be good to add tests with other timezones too. Might need to consider the value of user created at more closely to trigger edge cases.